packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts

```diff
@@ -0,0 +1,279 @@
+    minLength: IConfigRange;
+    rollbackOnlyClients: IConfigRange;
+    editOnlyClients: IConfigRange;
+    rollbackAndEditClients: IConfigRange;
+    opsPerRollbackRange: IConfigRange;
+}
+
+interface IClients {
+    clients: TestClient[];
+    rollbackOnlyCount: number;
+    editOnlyCount: number;
+    rollbackAndEditCount: number;
+}
+
+const allOperations: TestOperation[] = [
+    removeRange,
+    annotateRange,
+    insertAtRefPos,
+];
+
+export const debugOptions: IRollbackFarmConfig = {
+    minLength: { min: 2, max: 2 },
+    rollbackOnlyClients: { min: 0, max: 0 },
+    editOnlyClients: { min: 0, max: 0 },
+    rollbackAndEditClients: { min: 1, max: 1 },
+    opsPerRollbackRange: { min: 1, max: 15 },
+    opsPerRoundRange: { min: 1, max: 100 },
+    rounds: 5,
+    operations: allOperations,
+    incrementalLog: true,
+    growthFunc: (input: number) => input + 1,
+};
+
+export const defaultOptions: IRollbackFarmConfig = {
+    minLength: { min: 1, max: 512 },
+    rollbackOnlyClients: { min: 1, max: 1 },
+    editOnlyClients: { min: 1, max: 1 },
+    rollbackAndEditClients: { min: 1, max: 3 },
+    opsPerRollbackRange: { min: 1, max: 128 },
+    opsPerRoundRange: { min: 1, max: 128 },
+    rounds: 8,
+    operations: allOperations,
+    growthFunc: (input: number) => input * 2,
+};
+
+describe("MergeTree.Client", () => {
+    const opts =
+        defaultOptions;
+        // debugOptions;
+
+    // Generate a list of single character client names, support up to 69 clients
+    const clientNames = generateClientNames();
+
+    doOverRange(opts.minLength, opts.growthFunc, (minLength) => {
+        it(`RollbackFarm_${minLength}`, async () => {
+            const mt = random.engines.mt19937();
+            mt.seedWithArray([0xDEADBEEF, 0xFEEDBED, minLength]);
+
+            const clients: IClients = {
+                clients: [new TestClient()],
+                rollbackOnlyCount: opts.rollbackOnlyClients.min,
+                editOnlyCount: opts.editOnlyClients.min,
+                rollbackAndEditCount: opts.rollbackAndEditClients.min,
+            };
+            await addClients(
+                clients.clients,
+                opts.rollbackOnlyClients.min + opts.editOnlyClients.min + opts.rollbackAndEditClients.min,
+                clientNames);
+            const maxClientCount = 1 + opts.rollbackOnlyClients.max + opts.editOnlyClients.max
+                + opts.rollbackAndEditClients.max;
+
+            let seq = 0;
+            // eslint-disable-next-line no-constant-condition
+            while (true) {
+                seq = runMergeTreeOperationRunner(
+                    mt,
+                    seq,
+                    clients,
+                    minLength,
+                    opts,
+                );
+
+                if (clients.clients.length === maxClientCount) {
+                    break;
+                }
+
+                clients.clients.forEach((c) => c.updateMinSeq(seq));
```

Comment:
if you run this between rounds zamboni will run, in addition to running on remote ops and acts. beyond that you don't need to do anything specific